### PR TITLE
chore: remove trait EngineApiTreeHandler

### LINKED ISSUE #10123: remove EngineApiTreeHandler trait
this trait isn't actually used 

https://github.com/paradigmxyz/reth/blob/08158e444953b2bef998d50335967e3550fb403c/crates/engine/tree/src/tree/mod.rs#L334-L335

and we can unify the OP implementation here anyway so we can remove this and convert the functions into regular functions

### PR DESCRIPTION
**Pull request by mvares**
Closes: #10123

### CODE REVIEW

**Review by mattsse — CHANGES_REQUESTED** [review]
can we keep the function docs?

**Comment by mvares** [comment]
> can we keep the function docs?

Personally, I think the current documents made sense in the previous ones

File: crates/engine/tree/src/tree/mod.rs [hunk 1]
```diff
@@ -306,47 +306,6 @@ impl EngineApiTreeState {
...
-    /// [`ExecutionPayload`]. The Execution layer executes the transactions and validates the
-    /// state in the block header, then passes validation data back to Consensus layer, that
-    /// adds the block to the head of its own blockchain and attests to it. The block is then
-    /// broadcast over the consensus p2p network in the form of a "Beacon block".
-    ///
-    /// These responses should adhere to the [Engine API Spec for
-    /// `engine_newPayload`](https://github.com/ethereum/execution-apis/blob/main/src/engine/paris.md#specification).
-    ///
-    /// This returns a [`PayloadStatus`] that represents the outcome of a processed new payload and
-    /// returns an error if an internal error occurred.
```
Comment (mattsse) [inline crates/engine/tree/src/tree/mod.rs:328]: this I want to keep

File: crates/engine/tree/src/tree/mod.rs [hunk 2]
```diff
@@ -306,47 +306,6 @@ impl EngineApiTreeState {
...
-    ) -> Result<TreeOutcome<PayloadStatus>, InsertBlockFatalError>;
-
-    /// Invoked when we receive a new forkchoice update message. Calls into the blockchain tree
-    /// to resolve chain forks and ensure that the Execution Layer is working with the latest valid
-    /// chain.
-    ///
-    /// These responses should adhere to the [Engine API Spec for
-    /// `engine_forkchoiceUpdated`](https://github.com/ethereum/execution-apis/blob/main/src/engine/paris.md#specification-1).
-    ///
-    /// Returns an error if an internal error occurred like a database error.
```
Comment (mattsse) [inline crates/engine/tree/src/tree/mod.rs:342]: This I also want to keep

File: crates/engine/tree/src/tree/mod.rs [hunk 3]
```diff
@@ -306,47 +306,6 @@ impl EngineApiTreeState {
     }
 }
 
-/// The type responsible for processing engine API requests.
-pub trait EngineApiTreeHandler {
-    /// The engine type that this handler is for.
-    type Engine: EngineTypes;
-
-    /// Invoked when previously requested blocks were downloaded.
```
Comment (mattsse) [inline crates/engine/tree/src/tree/mod.rs:314]: this as well.

**Review by mattsse — CHANGES_REQUESTED** [review]

**Comment by mvares** [comment]
@mattsse, what you don't mentioned, can I remove?

**Review by mattsse — APPROVED** [review]

**Pull request merged**